js/Channel.js

```diff
@@ -3,79 +3,71 @@ const keccak256 = require('js-sha3').keccak256
 const ensure = require('./ensureTypes')
 
 const ChannelState = {
-	Unknown: 0,
-	Active: 1,
-	Expired: 2
+	Challenged: '115792089237316195423570985008687907853269984665640564039457584007913129639935',
+	Active: '0'
 }
 
 function Channel(args) {
-	this.creator = ensure.Address(args.creator)
-
+	this.leader = ensure.Address(args.leader)
+	this.follower = ensure.Address(args.follower)
+	this.guardian = ensure.Address(args.guardian)
 	this.tokenAddr = ensure.Address(args.tokenAddr)
-	this.tokenAmount = ensure.Uint256(args.tokenAmount)
-
-	this.validUntil = ensure.Uint256(args.validUntil)
-
-	this.validators = Array.isArray(args.validators) ? args.validators.map(ensure.Address) : []
-
-	this.spec = ensure.Bytes32(args.spec)
+	this.nonce = ensure.Bytes32(args.nonce)
 
-	Object.freeze(this.validators)
 	Object.freeze(this)
 
 	return this
 }
 
-Channel.prototype.hash = function(contractAddr) {
-	// contains contractAddr, so that it's not replayable
-	if (!contractAddr) throw new Error('contractAddr required')
+Channel.prototype.hash = function() {
 	return Buffer.from(
 		keccak256.arrayBuffer(
 			abi.rawEncode(
-				['address', 'address', 'address', 'uint256', 'uint256', 'address[]', 'bytes32'],
-				[
-					contractAddr,
-					this.creator,
-					this.tokenAddr,
-					this.tokenAmount,
-					this.validUntil,
-					this.validators,
-					this.spec
-				]
+				['address', 'address', 'address', 'address', 'bytes32'],
+				[this.leader, this.follower, this.guardian, this.tokenAddr, this.nonce]
 			)
 		)
 	)
 }
 
-Channel.prototype.hashHex = function(contractAddr) {
-	return `0x${this.hash(contractAddr).toString('hex')}`
+Channel.prototype.hashHex = function() {
+	return `0x${this.hash().toString('hex')}`
 }
 
 Channel.prototype.toSolidityTuple = function() {
-	// etherjs doesn't seem to want BN.js instances; hex is the lowest common denominator for web3/ethers
-	return [
-		this.creator,
-		this.tokenAddr,
-		`0x${this.tokenAmount.toString(16)}`,
-		`0x${this.validUntil.toString(16)}`,
-		this.validators,
-		this.spec
-	]
+	return [this.leader, this.follower, this.guardian, this.tokenAddr, this.nonce]
 }
 
 Channel.prototype.hashToSign = function(contractAddr, balanceRoot) {
 	// contains the channel hash (channelId), so that it's not replayable
-	return Channel.getSignableStateRoot(this.hashHex(contractAddr), balanceRoot)
+	return Channel.getSignableStateRoot(contractAddr, this.hashHex(), balanceRoot)
 }
 
 Channel.prototype.hashToSignHex = function(contractAddr, stateRoot) {
 	return `0x${this.hashToSign(contractAddr, stateRoot).toString('hex')}`
 }
 
+Channel.prototype.getResumeSignableRoot = function(challengeExpires) {
```

Comment:
this and getResumeSignableRootHex should be renamed to ResumeMessage - it's not a state root